### PR TITLE
agent-sample: support single fossilizer

### PR DESCRIPTION
**Pull request by t-bast**
This change is [Reviewable](https://reviewable.io/reviews/stratumn/indigo-js/223)

### CODE REVIEW

**Comment by simonvadee** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[packages/agent-sample/index.js, line 34 at r1](https://reviewable.io:443/reviews/stratumn/indigo-js/223#-L1q4bNEPuA35yucjtmz:-L1q4bNEPuA35yucjtn-:bj41ctn) ([raw file](https://github.com/stratumn/indigo-js/blob/39912ef0b234b83e45b830a2b09c8b8be5d3eb73/packages/agent-sample/index.js#L34)):*
> ```JavaScript
> const fossilizerHttpClients = [];
> // Create an HTTP fossilizer client to fossilize segments.
> // Assumes an HTTP fossilizer server is available on env.STRATUMN_FOSSILIZER_URL or http://fossilizer:6000.
> ```

just for my understanding, why do we always assume this ? Just like we assume a store is running on `store:5000` ?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/stratumn/indigo-js/223#-:-L1q4vj5sRVOhmo0N8rn:bnfp4nl)*

File: packages/agent-sample/index.js [hunk 1]
```diff
@@ -37,15 +47,29 @@ const agent = Agent.create({
...
-  // plugins you want to use
-  plugins: [plugins.agentUrl(agentUrl), plugins.actionArgs, plugins.stateHash]
-});
+agent.addProcess(
+  'message',
+  messageBoard,
+  storeHttpClient,
+  fossilizerHttpClients,
+  {
+    salt: process.env.STRATUMN_SALT || crypto.randomBytes(32).toString('hex'),
```
Comment (such) [inline packages/agent-sample/index.js]: we don't need a salt anymore

**Comment by t-bast** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[packages/agent-sample/index.js, line 34 at r1](https://reviewable.io:443/reviews/stratumn/indigo-js/223#-L1q4bNEPuA35yucjtmz:-L1q78m_SjDZUQvxC_R2:b-d6mg63) ([raw file](https://github.com/stratumn/indigo-js/blob/39912ef0b234b83e45b830a2b09c8b8be5d3eb73/packages/agent-sample/index.js#L34)):*
<details><summary><i>Previously, simonvadee wrote…</i></summary><blockquote>

just for my understanding, why do we always assume this ? Just like we assume a store is running on `store:5000` ?
</blockquote></details>

Yeah, it's what was done previously. It's not the most flexible (especially for multiple fossilizers) so we'll need to figure out something better (probably force using an environment variable to pass a list of fossilizer urls?)

---

*[packages/agent-sample/index.js, line 56 at r1](https://reviewable.io:443/reviews/stratumn/indigo-js/223#-L1YFp1jcCzUh45cvjMM-r1-56:-L1q7OZcp-1McqKcYpdw:b-kgnae8) ([raw file](https://github.com/stratumn/indigo-js/blob/39912ef0b234b83e45b830a2b09c8b8be5d3eb73/packages/agent-sample/index.js#L56)):*
<details><summary><i>Previously, such (Adrien Montfort) wrote…</i></summary><blockquote>

we don't need a salt anymore
</blockquote></details>

Good point, removing that

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/stratumn/indigo-js/223)*

**Comment by t-bast** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[packages/agent-sample/index.js, line 56 at r1](https://reviewable.io:443/reviews/stratumn/indigo-js/223#-L1YFp1jcCzUh45cvjMM-r1-56:-L1qA7nsUYRIbmsee6t9:b-896fix) ([raw file](https://github.com/stratumn/indigo-js/blob/39912ef0b234b83e45b830a2b09c8b8be5d3eb73/packages/agent-sample/index.js#L56)):*
<details><summary><i>Previously, t-bast (Bastien Teinturier) wrote…</i></summary><blockquote>

Good point, removing that
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/stratumn/indigo-js/223)*

**Comment by such** [comment]
Reviewed 4 of 4 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/stratumn/indigo-js/223)*

**Pull request merged**